packages/sparql-language-server/src/SparqlLanguageServer.ts

```diff
@@ -85,10 +87,36 @@ export class SparqlLanguageServer extends AbstractLanguageServer<
   }
 
   handleUpdateCompletionData(update: SparqlCompletionData) {
+    // `relationshipCompletionItems` and `typeCompletionItems` must be updated
+    // in 2 different scenarios:
+    // #1 - namespaces provided after relationshipBindings or typeBindings
+    // #2 - namespaces provided before relationshipBindings or typeBindings
+    // Otherwise you can find yourself with 1, both or neither reflecting the
+    // namespace prefixes based on the order the updates are processed, which is
+    // indeterminate.
     if (update.namespaceMap) {
       this.namespaceMap = namespaceArrayToObj(update.namespaceMap);
```

Comment:
One other comment: I would actually prefer renaming `update.namespaceMap` to `update.namespaces` since the type being received is an array, not a map (which is why it needs to be transformed _into_ a map). I think doing this would also mean that no changes would be required Studio-side, with the result that https://github.com/stardog-union/stardog-studio/pull/1135/files could be closed/abandoned, right?